### PR TITLE
Add Baremetal Topology Panel

### LINKED ISSUE #586: Create bare metal topology view in ESI ui
If you instantiate a UI (or use the esi.massopen.cloud dashboard), you can click on Network > Network Topology and see an... error screen. Search for the feature though, and you'll see what it's supposed to look like.

It would be really nice to use the same tool to implement a bare metal topology view. I think the exact same UI framework/tool can be used; all that's needed is to generate and pass in a different topology based one what we can from Ironic and ESI-Leap APIs. For example, something like:

Node > Node ports > Neutron ports > Network

There would be added complicated bits; for example, if the Neutron port is a trunk port we'd want to show trunk port information, and if it has a floating IP attached we'd want to show that as well.

### PR DESCRIPTION
**Pull request by ajamias**
In the dashboard, if you click on Network > Network Topology, it shows a graph topology of the networks and instances. This panel does not work for baremetal nodes which ESI uses. This adds a topology view of the baremetal nodes and their MAC address connections to network ports in addition to the already existing network topology.

### CODE REVIEW

**Comment by ajamias** [comment]
when you click on a node, a bubble should pop up. The buttons in the bubble may or may not work because I only worked toward the information they display, and not the actions they can take

File: esi_ui/content/baremetal_topology/templates/baremetal_topology/_graph_view.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      <span class="fa fa-th"></span>  {% trans "Toggle Network Collapse" %}
+    </label>
+  </div>
+    <button type="button" class="btn btn-default" id="center_topology">
+      <span class="fa fa-refresh"></span> {%trans "Center Topology" %}
+    </button>
+</div>
+<div id="topologyCanvasContainer" class="d3-container">
+  <div class="nodata">{% blocktrans trimmed %}
+    There are no networks, routers, or connected instances to display.
```
Comment (tzumainn) [inline esi_ui/content/baremetal_topology/templates/baremetal_topology/_graph_view.html]: Maybe change "connected instances" to "bare metal nodes"? Leave out the word "connected" - it looks like we display unattached nodes, which I think is 100% correct.

File: esi_ui/content/baremetal_topology/templates/baremetal_topology/client_side/_balloon_container.html [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            <a href="[[url]][[console]]" class="vnc_window">» [[open_console_label]]</a>
+            [[/console]]
+      </div>
+      [[#type]]
+      <div class="cell delete">
+        <button class="delete-device btn btn-danger btn-xs [[type]]"
+          data-type="[[type]]" data-device-id="[[id]]"
+          help_text="{% trans "This action cannot be undone." %}">
+            [[delete_label]]
+        </button>
```
Comment (tzumainn) [inline esi_ui/content/baremetal_topology/templates/baremetal_topology/client_side/_balloon_container.html]: I would delete both these actions.

**Review by tzumainn — CHANGES_REQUESTED** [review]
Some very minor comments.

Also, is it possible to expand the icon for nodes on the topology map to make them the primary focus? Don't worry about it if there's no obvious way to do it.

File: esi_ui/content/baremetal_topology/templates/baremetal_topology/_graph_view.html (hunk 1 above)
Comment (ajamias) [reply esi_ui/content/baremetal_topology/templates/baremetal_topology/_graph_view.html]: The nodes that get displayed are the ones with at least one MAC address assigned to it. Just like how `openstack esi node network list` only displays nodes with at least one MAC address assigned.

**Review by tzumainn — CHANGES_REQUESTED** [review]
Two more things I noticed:

* The topology map includes neutron ports that aren't currently in use (not associated with a baremetal port). Can we remove those?
* For the trunk port, I see that the subports connect to their respect networks; however, the trunk port does not. Can we add that missing connection?

**Comment by ajamias** [comment]
Should be fixed now

**Review by tzumainn — APPROVED** [review]
Perfect - thanks for doing this!

**Pull request merged**